src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -206,6 +206,9 @@ ClientRequest::process_op(instance_handle_t &ihref, Ref<PG> &pg)
       auto soid = m->get_hobj();
       if (!soid.is_head()) {
         fut = do_recover_missing(pg, soid.get_head());
+      } else if (auto snap = snap_need_to_recover(); snap) {
+        soid.snap = *snap;
+        fut = do_recover_missing(pg, soid);
```

Comment:
seems couldn't do rollback here, need find missing the specific snap needed.

Reply:
yeah, I'm working on that.